packages/template-mrt-reference-app/app/ssr.js

```diff
@@ -195,12 +195,60 @@ const cookieTest = async (req, res) => {
 }
 
 /**
- * Express handler that sets a response header and returns a JSON response with
- * diagnostic values.  Use ?name=test-name&value=test-value to set a response header.
+ * Express handler that sets response headers and returns a JSON response with
+ * diagnostic values.
+ * Use ?name1=test-name&value1=test-value&name2=test-name2&value2=test-value2
+ * to set two response headers.
  */
-const responseHeaderTest = async (req, res) => {
-    if (Object.hasOwn(req.query, 'name') && Object.hasOwn(req.query, 'value')) {
-        res.set(req.query.name, req.query.value)
+const responseHeadersTest = async (req, res) => {
+    let i = 0
+    let queryIdFound = true
+
+    do {
+        i = i + 1
+
+        const headerName = `name${i}`
+        const headerValue = `value${i}`
+        if (Object.hasOwn(req.query, headerName) && Object.hasOwn(req.query, headerValue)) {
+            res.set(req.query[headerName], req.query[headerValue])
+        } else {
+            queryIdFound = false
+        }
+    } while (queryIdFound)
+
+    res.json(jsonFromRequest(req))
+}
+
+/**
+ * Express handler that sets a multi value response headers and returns a JSON response with
+ * diagnostic values.  Use ?name=test-name&value1=test-value&value2=test-value2 to
+ * set a response header with multiple values.
+ */
+const responseMultiValueHeaderTest = async (req, res) => {
+    let i = 0
+    let queryIdFound = true
+    let multiHeaderValues = []
+
+    if (!Object.hasOwn(req.query, 'name')) {
+        res.json(jsonFromRequest(req))
+        return
+    }
+
+    do {
+        i = i + 1
+
+        const headerValue = `value${i}`
+
+        //Find all the values and then set as an array
+        if (Object.hasOwn(req.query, headerValue)) {
+            multiHeaderValues.push(req.query[headerValue])
+        } else {
+            queryIdFound = false
+        }
+    } while (queryIdFound)
+
+    if (multiHeaderValues.length > 0) {
+        res.set(req.query.name, multiHeaderValues)
     }
     res.json(jsonFromRequest(req))
 }
```

Comment:
I think you'd find the code works fine w/o the guards, but this is really just a nit:

```suggestion
const responseMultiValueHeaderTest = async (req, res) => {
    if (!req.query.name) {
        return res.json(jsonFromRequest(req));
    }

    const multiHeaderValues = [];
    let i = 1;

    while (req.query[`value${i}`]) {
        multiHeaderValues.push(req.query[`value${i}`]);
        i++;
    }

    if (multiHeaderValues.length) {
        res.set(req.query.name, multiHeaderValues);
    }

    return res.json(jsonFromRequest(req));
};
```